src/rnp/fficli.cpp

```diff
@@ -1838,6 +1838,69 @@ cli_rnp_revoke_key(cli_rnp_t *rnp, const char *key)
     return res;
 }
 
+bool
+cli_rnp_remove_key(cli_rnp_t *rnp, const char *key)
+{
+    std::vector<rnp_key_handle_t> keys;
+    if (!cli_rnp_keys_matching_string(rnp, keys, key, CLI_SEARCH_SUBKEYS)) {
+        ERR_MSG("Key matching '%s' not found.", key);
+        return false;
+    }
+    rnp_cfg_t *  cfg = cli_rnp_cfg(rnp);
+    bool         res = false;
+    bool         secret = false;
+    bool         primary = false;
+    uint32_t     flags = RNP_KEY_REMOVE_PUBLIC;
+    rnp_result_t ret = 0;
+
+    if (keys.size() > 1) {
+        ERR_MSG("Ambiguous input: too many keys found for '%s'.", key);
+        goto done;
+    }
+    if (rnp_key_have_secret(keys[0], &secret)) {
+        ERR_MSG("Error getting secret key presence.");
+        goto done;
+    }
+    if (rnp_key_is_primary(keys[0], &primary)) {
+        ERR_MSG("Key error.");
+        goto done;
+    }
+
+    if (secret) {
+        flags |= RNP_KEY_REMOVE_SECRET;
+    }
+    if (primary) {
+        flags |= RNP_KEY_REMOVE_SUBKEYS;
+    }
+
+    if (secret && !rnp_cfg_getbool(cfg, CFG_FORCE)) {
+        char reply[10];
+
+        fprintf(rnp->userio_out,
+                "Key %s has coresponding secret key. Do you really want to delete it  (y/N)? ",
+                key);
+        fflush(rnp->userio_out);
+
+        if (fgets(reply, sizeof(reply), rnp->userio_in) == NULL) {
```

Comment:
I see no good reason to duplicate this piece of code from `rnp_get_output_filename`. Let's put it into a single static (for now) function in this file. I suggest to pass a string for output to this function without the `  (y/N)?` tail, so that it adds a tail and processes the input as it would be a `get yes or no` function.

Reply:
Agree, we should have some single point to get input from the user, like `std::string cli_rnp_get_user_input()`. And maybe something on top of it like `cli_rnp_get_confirmation(const std::string &msg)`, which would repeat asking for Y/N/Yes/No.

Reply:
Thanks guys! Once this is done let’s merge it.

Reply:
@rrrooommmaaa please see updated code